### DISCUSSION TITLE
[Feature]: Wi-fi only option for foreground backup

### DISCUSSION THREAD

**Discussion by erlenddahl** [immich-app/immich#1598] [Feature Request]
### Feature detail

If I understand them correctly, issues #983 and #421 are both closed as the backup options work as intended. Therefore, this is a request for an additional feature: a wi-fi only option for the foreground upload would be very useful. I like the foreground upload, as that gives me a way of controlling when to upload. But when there is no wi-fi only option for it, that also means that I cannot use Immich at all when I am travelling, and that would be a shame. (I could rush into settings and deactivate the foreground upload, of course, but then I'd have to remember that each time). 

I think it would be very nice if the wi-fi only option worked for both types of upload.

(I discovered this today, when Immich used 7GB of my data quota because of some assets that failed and were retried a lot of times before I noticed. That may be another issue in itself, of course, but a wifi only option could have saved me.)

### Platform

Mobile App

**Comment by alextran1502** [comment]
~@zoodyy Fynn, what do you think about this? I assume we can do this on the Flutter side only right?~

**Comment by alextran1502** [comment]
~I don't use Android but I think we have this option for Android background backup already~

**Comment by alextran1502** [comment]
Ah this is foreground backup.

**Comment by erlenddahl** [comment]
Yep, that's right! I also believe you were right in your first observation, that this is something that must be added on the Flutter side. 

If you guys agree that it would be a good idea to add a wi-fi only option to the foreground backup as well (or a combined switch for foreground/background), I may be able to help. I don't know Flutter yet, but I know programming, so I could give it a try if you want.

**Comment by sgloutnikov** [comment]
WiFi only backups are something that I would love to see as well, getting Immich even closer to Google Photos parity. Maybe the [connectivity_plus](https://pub.dev/packages/connectivity_plus) can help? 

I'm starting to learn Flutter and Dart primarily because of the beautiful work done on this project. If time permits would love to contribute in the future. Keep up the great work!

**Comment by jschwalbe** [comment]
Would also love to see this feature added. Thanks!

**Comment by TassCat** [comment]
Bumping this, as I came here to request the exact same thing, for the exact same reason (on iOS mobile). Chewed up a ton of international data when traveling. Would love to see a WiFi only upload for foreground backup if you have the free cycles to implement it. Thanks for all of your hard work Alex & Team!

**Reply by alextran1502** [reply]
On iOS you can set this in Background App Refresh ![image](https://github.com/immich-app/immich/assets/27055614/841270d8-ba42-41be-879a-34998407466a)

**Reply by TassCat** [reply]
This would be for foreground app refresh when the _Automatic foreground backup_ option is on. The feature request would be to add an option to allow foreground backup only when connected to WiFi vs connected to WiFi and Cellular

**Reply by alextran1502** [reply]
@TassCat this is the limitation of iOS at the moment, from
What I remember we don't have the sole control over this beside using this phone's setting

**Reply by TassCat** [reply]
Maybe I'm misunderstanding this issue.  I'm looking to support an option to allow "wi-fi only" for foreground upload.  However I wouldn't know if foreground upload is related to backgroundapprefresh for ios.

I'm just looking to not overrun my international data allowance.  For reference, I also use NextCloud for iOS and it has an "Only use Wi-Fi connections" under Settings>Auto-Upload. Was hoping to help bring a similar feature to light.

**Reply by Batwam** [reply]
> On iOS you can set this in Background App Refresh ![image](https://github.com/immich-app/immich/assets/27055614/841270d8-ba42-41be-879a-34998407466a)

I had a look for this. Unfortunately (for me), this option is not Immich specific so it deactivates background refresh for ALL apps. I still want apps to access data, I just don't want massive files getting uploaded. I think that what people are looking for is a way to only upload photos/videos when using wifi. Other apps have this feature:
![image](https://github.com/immich-app/immich/assets/56710655/4022468e-0425-4772-9a92-74eb9cde8af8)

Even Dropbox allows it:
![image](https://github.com/immich-app/immich/assets/56710655/86d11287-85ed-4d0f-b7da-9c32b6fb6063)

The idea would be to be able to use Immich to browse the files but not unnecessarily use the data quota uploading files (especially videos). Is this possible? Ideally there should be a separate option for photos and videos like in the first example from the app which shall not be named.

**Comment by LouisRed** [comment]
Hi! 
I saw my thread was closed due to duplication with this one, but I wasn't referring to exactly the same thing. I was referring to the possibility of enabling automatic backup only when connected to a specific Wi-Fi network and not backing up in other Wi-Fi or mobile networks. I didn't see that specific specification in this thread, if I'm not mistaken. 
Thanks!

**Comment by Quentin1010** [comment]
Any news on that?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/
